extension/src/popup/components/signTransaction/Operations/KeyVal/index.tsx

```diff
@@ -471,16 +472,19 @@ export const KeyValueInvokeHostFn = ({
 
   function renderDetails() {
     switch (hostfn.switch()) {
+      case xdr.HostFunctionType.hostFunctionTypeCreateContractV2():
       case xdr.HostFunctionType.hostFunctionTypeCreateContract(): {
-        const createContractArgs = hostfn.createContract();
-        const preimage = createContractArgs.contractIdPreimage();
-        const executable = createContractArgs.executable();
+        const createContractArgs = getCreateContractArgs(hostfn);
+        const preimage = createContractArgs.contractIdPreimage;
+        const executable = createContractArgs.executable;
```

Comment:
noticing that we're not calling contractIdPreimage() and createContractArgs.executable() anymore. Is this an API change from the SDK? Wondering if this is a code smell that users who don't upgrade Freighter immediately when P22 hits might have issues

Reply:
no it's not an API change, I just added `getCreateContractArgs` helper that does the unpacking of all of those and the constructor args so they're just not called there anymore.